File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 0;
        int i = digits.length-1;
        do{
            digits[i]+= 1;
            if(digits[i]==10){
                digits[i] = 0;
                carry = 1;
            }else
                carry = 0;
            i--;
        }while(carry!=0 && i>=0);
        if(carry==1){
            int[] ans = new int[digits.length+1];
            ans[0] = 1;
            for(int j = 0;j<digits.length;j++)
                ans[j+1] = digits[j];
            return ans;
        }
        return digits;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits)-1,-1,-1):
            digits[i]+=carry
            if digits[i]>9:
                digits[i] = 0
                carry = 1
            else:
                carry = 0
                break
        if carry == 1:
            ans = []
            ans.append(1)
            for digit in digits:
                ans.append(digit)
            return ans
        return digits